rustup-init.sh

```diff
@@ -493,15 +493,27 @@ check_help_for() {
 
     case "$_arch" in
 
-        # If we're running on OS-X, older than 10.13, then we always
-        # fail to find these options to force fallback
         *darwin*)
         if check_cmd sw_vers; then
-            if [ "$(sw_vers -productVersion | cut -d. -f2)" -lt 13 ]; then
-                # Older than 10.13
-                echo "Warning: Detected OS X platform older than 10.13"
-                return 1
-            fi
+            case $(sw_vers -productVersion) in
+                10.*)
+                    # If we're running on OS-X, older than 10.13, then we always
+                    # fail to find these options to force fallback
+                    if [ "$(sw_vers -productVersion | cut -d. -f2)" -lt 13 ]; then
+                        # Older than 10.13
+                        echo "Warning: Detected OS X platform older than 10.13"
+                        return 1
+                    fi
+                    ;;
+                11.*)
```

Comment:
It *shouldn't* matter, but I'll point out that Big Sur [is _sometimes_ reported as 10.16](https://eclecticlight.co/2020/07/21/big-sur-is-both-10-16-and-11-0-its-official/). I doubt that will come into play for Rustup.

Reply:
Is this perhaps why you don't see the issue?

Reply:
It’s possible somehow, but running sw_vers *did* show `11...`

Reply:
I wonder if it's possibly related to running BigSur on M1 as opposed to on Intel?  I am running this on M1.
The 10.16 may have also been the version reported in beta releases, and might now have been cleaned up? (Just guessing here, really).

Reply:
This difference appears to be due to changes introduced in 01fd3a19774e97839f6b3c52075a491ea9479abd

The code changes otherwise seem to work fine.

Reply:
Regarding 10.16 vs 11.0, this seems to explain it: https://eclecticlight.co/2020/07/21/big-sur-is-both-10-16-and-11-0-its-official/

(Summary: It depends on whether the calling environment has `SYSTEM_VERSION_COMPAT=1` set, it seems)